lib/matplotlib/tests/test_colorbar.py

```diff
@@ -1034,8 +1034,9 @@ def test_colorbar_contourf_extend_patches():
         for ax, (extend, levels, hatches) in zip(axs, params):
             cs = ax.contourf(x, y, z, levels, hatches=hatches,
                              cmap=cmap, extend=extend)
-            subfig.colorbar(cs, ax=ax, orientation=orientation, fraction=0.4,
-                            extendfrac=0.2, aspect=5)
+            with pytest.warns(UserWarning, match="different Figure"):
+                subfig.colorbar(cs, ax=ax, orientation=orientation, fraction=0.4,
```

Comment:
Ah, subfigures are an edge case. IMHO we should not warn on different subfigures as long as the colorbar and mappable are in the same top-level figure. To that end, subfigures are more a layout element, and it is still ensured that mappable and colorbar are rendered alongside each other.

Reply:
I see, thanks for clarifying.

Reply:
Agreed. But if it's not useful, we should rewrite the test. Doesn't make sense to have a contrived test that needs handling of a warning because of its awkward structure.

Actually, looking at the test, why does it raise the warning? All operation seem to run on the
sub figures individually. AFAICS
- we create subplots in the sub figure
- we add a contourset to each Axes
- Add a colorbar for each contourset, stealing space from the associated Axes

Reply:
I think the test case warned because I was checking if `self.figure` is the same instance as `mappable.figure` in `fig.colorbar(mappable)`, which returned false when:
- Both `self.figure` and `mappable.figure` are figure instances, but not the same one.
- One of them is figure and the other subfigure.

In the latest commit this should be fixed. When `self.figure` or `mappable.figure` is a subfigure, I use the associated figure for comparison.

Reply:
Oh, also, if i'm getting your question correctly, the test case failed in the previous run becasue it did not warn while `with pytest.warns(...)` is to make sure it actually warns.